### ISSUE TITLE
docker crashes with no clear error when settings.json is empty

### ISSUE THREAD

**Issue by markg85** [moby/moby#41268]
Hi,

`Docker version 19.03.11, build 42e35e6`

If settings.json is empty.
Really empty!
Like:
` `

Then docker fails to start and fails to provide any meaningful message as to why it can't start.

At the moment there are 2 ways to fix this.
1. Make settings.json contain an empty json object:
`{}`

2. Remove settings.json

Cheers,
Mark

**Comment by AkihiroSuda** [comment]
You mean /etc/docker/daemon.json ?

**Comment by markg85** [comment]
> You mean /etc/docker/daemon.json ?

Yes.
Sorry, my mistake.

I edited it to reflect that.

**Comment by thaJeztah** [comment]
I made a change for a similar situation in the CLI recently https://github.com/docker/cli/pull/2599

We should look carefully if it's safer to "ignore" or to error with a clear error message though; an existing file (but empty) can be the result of something that went wrong, and having an error, instead of silently ignoring that situation could be useful.

If we decide to keep it as an error condition, I agree that the error could be more descriptive;

```bash
mkdir /etc/docker
touch /etc/docker/daemon.json
dockerd
# unable to configure the Docker daemon with file /etc/docker/daemon.json: EOF
```

**Comment by markg85** [comment]
@thaJeztah that's not what i get.
Granted, i run docker via systemd. There is simply get an error that systemd killed the process because docker was in a restart loop.

In other words, nothing about daemon.json **at all**.

**Comment by thaJeztah** [comment]
Yes, systemd's output for failed services is not very useful most of the times unfortunately; most of the time it only shows "it failed", without any detail _why_.

You should be able to get more details from the daemon logs (for example, `sudo journalctl -n 200 -xu docker.service` to get the last 200 log entries)
